mediaengine.go

```diff
@@ -432,3 +293,44 @@ func (m *MediaEngine) collectStats(collector *statsReportCollector) {
 		collector.Collect(stats.ID, stats)
 	}
 }
+
+// TODO Sean-Der, everything below is not finished
+
+// Update the MediaEngine from a remote description
+func (m *MediaEngine) updateFromRemoteDescription(desc sdp.SessionDescription) error {
+	for _, media := range desc.MediaDescriptions {
+		switch {
+		case !m.negotiatedAudio && strings.EqualFold(media.MediaName.Media, "audio"):
+			m.negotiatedAudio = true
+		case !m.negotiatedVideo && strings.EqualFold(media.MediaName.Media, "video"):
+			m.negotiatedVideo = true
+		default:
+			continue
+		}
+
+		// * convert media.Attributes in []RTPCodecParameters
+		// * Update media.videoCodecs (or media.audioCodecs)
+		//   - set negotiated=true
+		//   - set the proper payloadtype
+		//   - set the proper fmtp
```

Comment:
i think it should also:
- set proper extension ids, and remove extensions not present in sdp
- remove all feedback not present in the sdp
- remove all codecs not present in sdp

this method is strongly related to addTransceiverSDP, as it will determine whether the returned sdp is correct or not (the answer sdp must contain the same payload type/extension id/feedback mechanisms)

Reply:
👍 hopefully will finish this all tonight. I can't wait to merge this and then your Interceptor PR and just tag `v3.0.0`

We can follow up quickly with `v3.0.1`, but this was just so much more work then I anticipated.

Reply:
One more thing that came to mind is renegotiation: what happens, if we support opus codec, but there is no audio track in the original sdp, so we remove it from MediaEngine, but then there is a renegotiation and an opus track is added (is that possbile? I have never done re-negotiation). Maybe instead of setting negotiated: true parameters, you could save the negotiated codecs in a separate slice:
```
type MediaEngine struct {
	videoCodecs           []RTPCodecParameters
	audioCodecs           []RTPCodecParameters
	negotiatedVideoCodecs []RTPCodecParameters
	negotiatedAudioCodecs []RTPCodecParameters
}
```

Reply:
That is a much better design! Before I added `negotiated` as an entry for Headers/Codecs.

I am going to instead have a seperate list for negotiated Headers/Codecs. It will be much easier to understand the state of everythinhg.

Reply:
Done!